builder/builder-source-git.c

```diff
@@ -417,75 +401,80 @@ builder_source_git_download (BuilderSource  *source,
 static gboolean
 git_extract_submodule (const char     *repo_location,
                        GFile          *checkout_dir,
+                       const char     *revision,
                        BuilderContext *context,
                        GError        **error)
 {
-  g_autofree char *submodule_status = NULL;
-
-  if (!git (checkout_dir, &submodule_status, error,
-            "submodule", "status", NULL))
-    return FALSE;
+  g_autoptr(GKeyFile) key_file = g_key_file_new ();
+  g_autofree gchar *submodule_data = NULL;
+  g_autofree gchar **submodules = NULL;
+  g_autofree gchar *gitmodules = g_strconcat (revision, ":.gitmodules", NULL);
+  gsize num_submodules;
 
-  if (submodule_status)
+  if (git (checkout_dir, &submodule_data, NULL, "show", gitmodules, NULL))
     {
```

Comment:
In that case the `git` function returns `FALSE`, and we do not enter the subsequent block.

However, running that ` git show` command will print an error message if there aren't any submodules:

```
fatal: Path '.gitmodules' does not exist in '5367783d35897e2d2a832ba7805ec9f050997a37'
```

That's a bit ugly, but it is completely harmless. However it might confuse people.

So I'm not sure what to do about this.